Kernel/Process.cpp

```diff
@@ -657,6 +718,18 @@ void Process::finalize()
     m_fds.with_exclusive([](auto& fds) { fds.clear(); });
     m_tty = nullptr;
     m_executable.with([](auto& executable) { executable = nullptr; });
+    m_attached_jail.with([](auto& jail) {
+        if (jail) {
+            jail->attach_count().with([&](auto& attach_count) {
```

Comment:
can we make this entire logic here a member function on Jail?

`Jail::detach()`

Then we don't need to touch its private members from Process.